docker: fix systemd service

mouyase opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Topic Description
-----------------
docker:  fix systemd service

Drop `/etc/conf.d/docker` and to allow Docker honor user settings from the default configuration file `/etc/docker/daemon.json`.
Copy docker.service file from Debian.

<!-- Please input topic description here. -->

Package(s) Affected
-------------------

<!-- Please list all package(s) affected by this topic here. -->
docker

Security Update?
----------------

<!-- If this topic contains security update(s), please uncomment "Yes,"
     mark with the `security` and `priority` labels, and make sure to mark your commits to relevant issue numbers.

     Please see GitHub's documentation on "Linking a pull request to an issue":

     https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

<!-- Yes - Issue Number: ISSUENUMBER -->
<!-- No -->
No

<!-- Please uncomment the "Build Order" section if your topic affects more than one package. -->

<!--
Build Order
-----------
-->

<!-- Please describe in what order maintainers should build this pull request. You can use the following template, use space to separate packages. -->

<!--
```
pkg1 pkg2 pkg3 ...
```
-->

Test Build(s) Done
------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [ ] AMD64 `amd64`
- [ ] AArch64 `arm64`
- [ ] LoongArch 64-bit `loongarch64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for secondary ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

<!-- Maintainers should review file changes and make sure that the change(s) made complies with our [Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/) -->

<!-- Maintainers and users may now test the packages in this topic and, once user/maintainer feedback indicates that the update(s) work as expected and find its quality satisfactory,
     another maintainer may now review this pull request and mark it as Approved. After which, the maintainer will build affected package(s) and upload them to the `stable` repository. -->

MingcongBai commented on app-containers/docker/autobuild/overrides/usr/lib/systemd/system/docker.service:
```diff
@@ -1,38 +1,46 @@
 
 [Install]
-WantedBy=multi-user.target
+WantedBy=multi-user.target
```
Missing newline at EOF.

MingcongBai replied:
Bump REL in `spec` to mark this package update as a revision.

MingcongBai commented on app-containers/docker/autobuild/overrides/usr/lib/systemd/system/docker.service:
```diff
@@ -1,38 +1,46 @@
 [Unit]
```
Where did this service come from (you wrote it, or from another distribution)? If the latter, please note the source of the service file in your Git commit message.

MingcongBai commented on app-containers/docker/autobuild/overrides/etc/conf.d/docker:
This will likely need a maintscript handler. I will handle it on my end.

MingcongBai commented on app-containers/docker/autobuild/postinst:
Seems unnecessary to do it three times... When is this most appropriate?

mouyase replied:
https://man.archlinux.org/man/extra/dpkg/dpkg-maintscript-helper.1.en#Removing_a_conffile

>Current implementation: the preinst checks if the conffile has been modified, if yes it's left on place otherwise it's renamed to old-conffile.dpkg-remove. On configuration, the postinst removes old-conffile.dpkg-remove and renames old-conffile to new-conffile if old-conffile is still available. On abort-upgrade/abort-install, the postrm renames old-conffile.dpkg-remove back to old-conffile if required.

So must do it three times.

MingcongBai replied:
I see, my bad.

mouyase committed:
docker:  fix systemd service

Drop `/etc/conf.d/docker` and to allow Docker honor user settings from the default configuration file `/etc/docker/daemon.json`.
Copy docker.service file from Debian.
